### PR TITLE
fix(core): Fix job executor stuck indefinitely

### PR DESCRIPTION
**Pull request by aufarg**
This commit introduces a timeout value for streaming job to prevent the agent thread to stuck indefinitely when something goes wrong with the job.

Fixes: https://github.com/spinnaker/spinnaker/issues/6288

### CODE REVIEW

**Comment by aufarg** [comment]
I don't think there's a test file for this class, should I add a new test file for `JobExecutorLocal`?

**Comment by aufarg** [comment]
Right now, I'm using the existing timeout value so if user haven't configure timeout yet, they will notice a change in behavior where work that can actually complete with more than 10m will start to time out.
If this is not desired, we can make a separate timeout for streaming jobs. Let me know which one you think is better!

**Comment by ajordens** [comment]
Can you rebase this PR, it's hard to follow with the other commits pulled in.

**Comment by aufarg** [comment]
Done

**Comment by mitali-sahoo** [comment]
@aufarg @ajordens  I am facing the similar caching issue on spinnaker. We are hosting v1.20.7 spinnaker version currently. Could you please consider merging the changes to the mentioned version too ?

**Comment by rampatnaik** [comment]
weare facing the similar caching issue on spinnaker. We are hosting v1.20.7 spinnaker version currently. Could you please consider merging the changes to the mentioned version too ?

**Comment by sethev** [comment]
This can also cause monitorDeploy to hang indefinitely if this occurs during a deploy task: https://github.com/spinnaker/spinnaker/issues/6410#issuecomment-923440166

**Comment by sethev** [comment]
@aufarg want to update and resolve the conflict?

**Review by deverton — APPROVED** [review]

**Review by dbyron-sf — APPROVED** [review]

**Review by jkschneider — APPROVED** [review]
